libraries/botbuilder-dialogs-adaptive/src/actions/setProperties.ts

```diff
@@ -0,0 +1,45 @@
+import { Dialog, DialogContext, DialogTurnResult, DialogConfiguration } from 'botbuilder-dialogs';
+import { ExpressionEngine } from 'botframework-expressions';
+
+export class PropertyAssignment {
+    public property: string;
+    public value: string;
+}
+
+export interface SetPropertiesConfiguration extends DialogConfiguration {
+    assignments?: PropertyAssignment[];
+}
+
+export class SetProperties<O extends object = {}> extends Dialog<O> {
+
+    public static declarativeType = 'Microsoft.SetProperties';
+
+    /**
+     * Additional property settings as property/value pairs.
+     */
+    public assignments: PropertyAssignment[] = [];
+
+    public configure(config: SetPropertiesConfiguration): this {
+        return super.configure(config);
+    }
+
+    public async beginDialog(dc: DialogContext, options?: O): Promise<DialogTurnResult> {
+        for (let i = 0; i < this.assignments.length; i++) {
+            const assignment = this.assignments[i];
+            const valExp = new ExpressionEngine().parse(assignment.value);
```

Comment:
>const valExp = new ExpressionEngine().parse(assignment.value); [](start = 12, length = 62)

You could potentially cache these on first access. Look at SwitchCondition for an example.

Reply:
> > const valExp = new ExpressionEngine().parse(assignment.value); [](start = 12, length = 62)
> 
> You could potentially cache these on first access. Look at SwitchCondition for an example.

Cached parsed assignment value expressions.